### ISSUE TITLE
Immediate Segmentation Fault with library after init under linux - Stacksize

### ISSUE THREAD

**Issue by romw** [BOINC/boinc-ng#1163]
**Reported by tstrunk on 23 Apr 42474430 18:25 UTC**
When building an application with the current boinc api lib and running it live (that is with shared mem), the timer thread segfaults on Ubuntu 11.10.

The culprit is the timer thread, which dies after TIMER_PERIOD seconds are over (first time timer_handler is called). I think the problem is that the child thread exhausts its stacksize. I can attribute the crash to this changeset (25659):
http://boinc.berkeley.edu/trac/changeset/25659/boinc/trunk/boinc/api/boinc_api.cpp

On Ubuntu MAXPATHLEN is 4096. Setting any of the fields changed in the changeset to MAXPATHLEN will produce a segfault. Even raising only abspath[to abspath[3096](1024]) produces said segfault.

I'd like to increase the threads stacksize in valgrind, but you can only increase the main thread stacksize there, not the childs. This is one of the error messages:
==24957== Can't extend stack to 0x40273c0 during signal delivery for thread 2:
==24957==   too small or bad protection modes

This is the test application, it inits and then sleeps forever and either segfaults or doesn't, depending on MAXPATHLEN.

```

#include <iostream>
#include <diagnostics.h>
#include <boinc_api.h>
int main(int argc, const char* argv [])
{
    boinc_init_diagnostics( BOINC_DIAG_TRACETOSTDERR|BOINC_DIAG_HEAPCHECKENABLED);
    int rc = boinc_init();
    for (unsigned int i = 0; i < 10000 ;)
    {
        sleep(3000);
    }
}

```

**Comment by romw** [comment]
**Commented by tstrunk on 24 Nov 42474524 01:17 UTC**
Revision [25659] might have been the wrong one. Although I do not get any errors anymore with the testcode, when reverting it, I get a similar valgrind errors, with our complete binary.

**Comment by romw** [comment]
**Commented by davea on 1 May 42474577 18:58 UTC**
(In [25781]) - API, Linux: increase timer thread stack size from 16K to 32K.
    Apparently new use of MAXPATHLEN can cause overflow.
    Fixes #1191

**Comment by romw** [comment]
**Commented by tstrunk on 29 Sep 42487157 00:44 UTC**
This fixed all the crashes I was experiencing.
Thanks!
